wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -162,6 +166,20 @@ getAccounts
     => Maybe (CId Wal) -> m [CAccount]
 getAccounts = getAccountsIncludeUnready False
 
+getAccountsPaginate
+    :: MonadWalletWebMode m
+    => Maybe (CId Wal) -> Maybe (CId Addr) -> m [CAccount]
+getAccountsPaginate mwid mcid = do
+    accounts <- getAccounts mwid
```

Comment:
Cannot we simply access the full list of accounts for an address in the `WalletStorage`, and do a Map lookup rather than getting all the addresses followed by a filter afterwards?

Reply:
Oh, to do it correctly we need to check both db and mempool, since where could be addresses which are made by us just recently and not yet in blocks (compare with https://goo.gl/NY3MUJ). Thus it won't be the "most easy solution", as you had said :sweat_smile:.
But I guess, it's worth going the way you mention anyway, right?

Reply:
I think that's a trade-off. We could allow this endpoint to return a slight (but self healing) stale view of the world, considering is idempotent and is being accessed via a GET. If we relax this invariant, we can do it "the easy and efficient way". Whether or not we can relax this invariant, I guess it really depends from the use case and the expectations from this endpoint, which we should probe for 😉

Reply:
Yes, this makes sense. For me it looks like this endpoint would become primary one for API users along with `api/accounts` and others.
On the other hand, if we have several endpoints which return addresses, returning different views on them (from temporal point of time) may be confusing. :thinking:

Reply:
> For me it looks like this endpoint would become primary one for API users along with api/accounts and others.

I think the primary target here are exchanges, and if we instruct them that some addresses might not show up if not yet accepted in blocks, it could be fine (or maybe not, that's why we need to probe :) )

> On the other hand, if we have several endpoints which return addresses, returning different views on them (from temporal point of time) may be confusing.

On the other hand, the state of the blockchain and the state of the wallet changes with time, so maybe that's not so absurd 😉

Reply:
Gotcha. I'm on it :+1: